src/AwkwardPythonCallExt.jl

```diff
@@ -0,0 +1,116 @@
+module AwkwardPythonCallExt
+using PythonCall
+using JSON
+import AwkwardArray
+
+function AwkwardArray.convert(layout::AwkwardArray.Content)::Py
+    form, len, containers = AwkwardArray.to_buffers(layout)
+
+    py_buffers = Dict{String,Any}()
+
+    for (key, buffer) in containers
+        py_buffers[key] = pyimport("numpy").asarray(buffer, dtype = pyimport("numpy").uint8)
+    end
+
+    pyimport("awkward").from_buffers(form, len, py_buffers)
+end
+
+function AwkwardArray.convert(array::Py)::AwkwardArray.Content
+    form, len, _containers = pyimport("awkward").to_buffers(array)
+    containers = pyconvert(Dict, _containers)
+
+    julia_buffers = Dict{String,Vector{UInt8}}()
+
+    for (key, buffer) in containers
+        julia_buffers[key] = reinterpret(UInt8, buffer)
+    end
+
+    AwkwardArray.from_buffers(
+        pyconvert(String, form.to_json()),
+        pyconvert(Int, len),
+        julia_buffers,
+    )
+end
+
+# rule functions
+function pyconvert_rule_awkward_array_primitive(::Type{AwkwardArray.PrimitiveArray}, x::Py)
+    array = AwkwardArray.convert(x)
+    return PythonCall.pyconvert_return(array)
+end
```

Comment:
These rule functions should return `pyconvert_unconverted()` if the conversion was not possible.

Reply:
Is that distinct from encountering an Exception and raising it?

Reply:
> These rule functions <strike>should</strike> **must** return `pyconvert_unconverted()` if the conversion was not possible.

https://github.com/JuliaPy/PythonCall.jl/blob/13f596d6a7d60ef7bfcee2d538cd895f59826d95/src/Convert/pyconvert.jl#L38C1-L45